### ISSUE TITLE
Decimal formatting

### ISSUE THREAD

**Issue by stephanstapel** [mnadel/metrics.net.influxdb#2]
Hi,

the library currently does not work in regions with , (comma) as a decimal separator.
I corrected this for now by modifying InfluxDbRecord.cs with:
```
private NumberFormatInfo _nfi = new NumberFormatInfo()
{
  NumberDecimalSeparator = "."
};
```
and modifying value compilation:
```
foreach (var pair in columns.Zip(data, (col, dat) => new { col, dat })) {
  fieldKeypairs.Add (string.Format ("{0}={1}", Escape (pair.col), ((double)pair.dat).ToString(_nfi)));
}
```

I guess there are better alternatives since one can't guarantee that the value object contains a double.
Comment welcome!

**Comment by mnadel** [comment]
Good catch, thanks. Not a fan of the cast to a double, though. Maybe all numeric values are doubles, but not all values are numeric. Histograms, for example, will keep track of [user values](https://github.com/etishor/Metrics.NET/wiki/Histograms#keeping-track-of-user-values) which are strings.

How's this look?

https://github.com/mnadel/metrics.net.influxdb/pull/3

The dict exists to find a formatter in constant time, and ints are handled differently from floats (InfluxDB wants integer types to be suffixed with an "i".)

**Comment by stephanstapel** [comment]
Your solution looks really nice, good approach!

How about generating

```
CultureInfo.CreateSpecificCulture ("en-US")
```

once (either private oder static) instead of generating it for every conversion?

**Comment by mnadel** [comment]
Thanks for reviewing. I've made your suggested change and pushed to the branch. I'll test against InfluxDB this weekend and if all looks good will release within a few days. If you have a chance to run some integration testing in your locale, it'd be much appreciated! Cheers.

**Comment by stephanstapel** [comment]
I deployed the new version into one of my productive applications. So far it looks great, I'll let you know in a few days if this stays.

btw: do you know if Metrics.net is stlil alive? It looks a bit sad to see all these pull requests and open issues.

**Comment by mnadel** [comment]
Great, thanks!

I'm not sure, but I'm with you -- it does appear dead. I originally contributed this integration directly to metrics.net as a PR, but eventually published it separately to Nuget.org when the PR went unacknowledged.

Curious -- what version of metrics.net are you using in production?

**Comment by stephanstapel** [comment]
I'm using the NuGet version 0.2.16.0 (https://www.nuget.org/packages/Metrics.NET/). I checked all the forked repositories but found that none of them compiled all the enhancements and fixes including yours.
I sent a message to etishor some minutes ago, https://github.com/etishor/Metrics.NET/issues/122. Hope that he answers.

I also decided to implement a wrapper into my applications to make me independent of Metrics.net - just in case. And fix the Gauge value-as-a-function implementation annoyance.
